src/api/Signals.js

```diff
@@ -26,44 +27,13 @@ export default class Signals {
   addSignal(doc: SearchDocument, type: string = 'click', weight: number = 1) {
     if (doc.signal) {
       const uri = `${this.baseUri}/rest/signals/add`;
-      const headers = new Headers({
-        Accept: 'application/json',
-        'Content-Type': 'application/json',
-      });
       const updatedSignal = Object.assign({}, doc.signal, { type, weight });
-      const body = JSON.stringify(updatedSignal);
-      const params = {
-        method: 'POST',
-        headers,
-        body,
-        credentials: 'include',
-      };
-      const fetchRequest = new Request(uri, params);
-
-      fetch(fetchRequest).then(
-        (response: Response) => {
-          if (response.ok) {
-            response.json().then(() => {
-            }).catch((error: any) => {
-              // Catch errors from converting the response's JSON
-              console.log('Failed to submit signal', updatedSignal, error);
-            });
-          } else {
-            // The request came back other than a 200-type response code
-            const message = response.statusText ?
-              `${response.statusText} (error code ${response.status})` :
-              `Unknown error of type ${response.status}`;
-            console.log('Failed to submit signal', updatedSignal, message);
-          }
-        },
-        (error: any) => {
-          // Catch network-type errors from the main fetch() call
+      const callback = (response: any | null, error: string | null) => {
+        if (!error) {
           console.log('Failed to submit signal', updatedSignal, error);
```

Comment:
This seems to be logging the failure if there's no error... is that what we want?

Reply:
This seems to be incorrect. We should log the errors if there are any. I would remove the negation in that case.